clients/python/tensorzero/tensorzero.pyi

```diff
@@ -219,6 +220,36 @@ class TensorZeroGateway(BaseTensorZeroGateway):
         :param datapoint_id: The ID of the datapoint to delete.
         """
 
+    def list_datapoints(
+        self,
+        *,
+        dataset_name: str,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
```

Comment:
For clarity to end users, don't make these optional and rather do:

```python
limit: int = 100,
offset: int = 0,
```

Trickle down to other pieces of this function.

Reply:
how do we handle this for HTTP? I don't like mandatory search params

Reply:
fair